spec/acceptance/mongos_spec.rb

```diff
@@ -0,0 +1,125 @@
+require 'spec_helper_acceptance'
+
+describe 'mongodb::mongos class' do
+
+  shared_examples 'normal tests' do |tengen|
+    if tengen
+      case fact('osfamily')
+      when 'RedHat'
+        package_name = 'mongodb-org-mongos'
+        service_name = 'mongos'
+        config_file  = '/etc/mongos.conf'
+      when 'Debian'
+        package_name = 'mongodb-org-mongos'
+        service_name = 'mongos'
+        config_file  = '/etc/mongos.conf'
+      end
+    else
+      case fact('osfamily')
+      when 'RedHat'
+        package_name = 'mongodb-server'
+        service_name = 'mongos'
+        config_file  = '/etc/mongos.conf'
+      when 'Debian'
+        package_name = 'mongodb-server'
+        service_name = 'mongods'
```

Comment:
If these options are all the same between RedHat and Debian then there's probably no need to use cases.